packages/cozy-client/src/StackLink.js

```diff
@@ -77,11 +80,26 @@ export default class StackLink extends CozyLink {
     this.stackClient = null
   }
 
-  request(operation, result, forward) {
-    if (operation.mutationType) {
-      return this.executeMutation(operation, result, forward)
+  async request(operation, result, forward) {
+    if (this.isOnline && !(await this.isOnline())) {
+      return forward(operation)
     }
-    return this.executeQuery(operation)
+
+    try {
+      if (operation.mutationType) {
+        return await this.executeMutation(operation, result, forward)
```

Comment:
Why did you add the `await`?

Reply:
Without the await, the `try...catch` would not be able to catch anything executed inside the `executeMutation()` method.

This is because we directly return the Promise, so the `try...catch` will directly release the flow control without waiting for the `executeMutation` result.

If we add an `await` then the `try...catch` will continue its action until the Promise is fulfilled.

Reply:
Very clear, thank you